### PR TITLE
Feature/#80 allow user to restore password

### PR DESCRIPTION
**Pull request by ZVARKO**

### CODE REVIEW

File: WebAPI/Controllers/UserController.cs [hunk 1]
```diff
@@ -123,5 +123,55 @@ public IActionResult Renew([FromBody] RenewModel body)
...
+        {
+            body.Email = body.Email.ToLower();
+            var user = await _userService.GetUserByEmailAsync(body.Email);
+            if (user == null)
+            {
+                _logger.LogInformation($"Invalid password reset attemp. User {body.Email} doesn't exist.");
+                return ErrorResponse($"Používateľ {body.Email} neexistuje.");
+            }
+
+            if (!user.EmailConfirmed)
```
Comment (marektavac) [inline WebAPI/Controllers/UserController.cs]: What about if I lost my confirmation email and I don't remember my password?

File: WebAPI/Controllers/UserController.cs [hunk 2]
```diff
@@ -123,5 +123,55 @@ public IActionResult Renew([FromBody] RenewModel body)
...
+            {
+                _logger.LogInformation($"Invalid password reset attemp. User {body.Email} didn't confirm email address.");
+                return ErrorResponse($"Najskôr prosím potvrď svoju emailovú adresu.");
+            }
+
+            string token = await _userService.GeneratePasswordResetTokenAsync(user);
+            token = Uri.EscapeDataString(token);
```
Comment (marektavac) [inline WebAPI/Controllers/UserController.cs]: EscapeDataString is called everytime after GenerateToken, can be part of function above?

Comment (ZVARKO) [reply WebAPI/Controllers/UserController.cs]: done

File: WebAPI/Controllers/UserController.cs (hunk 1 above)
Comment (ZVARKO) [reply WebAPI/Controllers/UserController.cs]: User can resend register confirmation email. A button on login page appears when he tries to log in without having email confirmed.

**Pull request merged**